lib/pf/vlan/pool.pm

```diff
@@ -0,0 +1,154 @@
+package pf::vlan::pool;
+
+=head1 NAME
+
+pf::vlan::pool - Object oriented module for VLAN isolation oriented functions
+
+=head1 SYNOPSIS
+
+The pf::vlan::pool module contains the functions necessary for the VLAN Pool.
+
+=cut
+
+use strict;
+use warnings;
+
+use Log::Log4perl;
+
+use pf::config;
+use pf::util;
+use pf::log();
+
+use pf::node;
+
+use Number::Range;
+
+=head1 SUBROUTINES
+
+Warning: The list of subroutine is incomplete
+
+=cut
+
+=head2 new
+
+Constructor.
+
+=cut
+
+sub new {
+    my $logger =  pf::log::get_logger();
+    $logger->debug("instantiating new pf::vlan::pool object");
+    my ( $class, %argv ) = @_;
+    my $this = bless {}, $class;
+    return $this;
+}
+
+sub getVlanPool {
+    my ($self, $vlan, $switch, $ifIndex, $mac, $node_info, $connection_type, $user_name, $ssid, $radius_request, $role) = @_;
+    my $logger =  pf::log::get_logger();
+
+    return $vlan if $self->rangeValidator($vlan);
+    my $range = Number::Range->new($vlan);
+
+    $vlan = $self->getRoundRobin($mac, $role, $range);
+    return $vlan;
+}
+
+=head2 rangeValidator
+
+Validate the range definition
+Should be something like that 20..23 or 20..23,27..30
+
+=cut
+
+sub rangeValidator {
+    my ($self, $range) =@_;
+    my $rangesep = qr/(?:\.\.)/;
+    my $sectsep  = qr/(?:\s|,)/;
```

Comment:
I am not sure we should accept spaces as section separators.
It may be much easier in the end to just ignore spaces.
That could prevent a whole class of hard to debug errors (typos etc.)